stdlib/universe/aggregate_window.go

```diff
@@ -0,0 +1,951 @@
+	}
+
+	windowNode := aggregateNode.Predecessors()[0]
+	windowSpec := windowNode.ProcedureSpec().(*WindowProcedureSpec)
+	if !a.isValidWindowSpec(windowSpec) {
+		return node, false, nil
+	}
+	parentNode := windowNode.Predecessors()[0]
+
+	parentNode.ClearSuccessors()
+	newNode := plan.CreateUniquePhysicalNode(ctx, "aggregateWindow", &AggregateWindowProcedureSpec{
+		spec:      windowSpec,
+		aggregate: aggregate,
+		valueCol:  valueCol,
+		useStart:  useStart,
+	})
+	parentNode.AddSuccessors(newNode)
+	newNode.AddPredecessors(parentNode)
+	return newNode, true, nil
+}
+
+func (a AggregateWindowRule) isValidWindowInfSpec(spec *WindowProcedureSpec) bool {
+	return spec.TimeColumn == execute.DefaultTimeColLabel &&
+		spec.StartColumn == execute.DefaultStartColLabel &&
+		spec.StopColumn == execute.DefaultStopColLabel &&
+		!spec.CreateEmpty ||
+		spec.Window.Every == infinityVar.Duration()
+}
+
+func (a AggregateWindowRule) isValidDuplicateSpec(spec *SchemaMutationProcedureSpec) (useStart, ok bool) {
+	if len(spec.Mutations) != 1 {
+		return false, false
+	} else if s, ok := spec.Mutations[0].(*DuplicateOpSpec); !ok {
+		return false, false
+	} else {
+		if s.As != execute.DefaultTimeColLabel {
+			return false, false
+		}
+
+		switch s.Column {
+		case execute.DefaultStartColLabel:
+			useStart = true
+		case execute.DefaultStopColLabel:
+			useStart = false
+		default:
+			return false, false
+		}
+	}
+	return useStart, true
+}
+
+func (a AggregateWindowRule) isValidAggregateSpec(spec plan.ProcedureSpec) (aggregateWindow, string, bool) {
+	switch spec.Kind() {
+	case CountKind:
+		aggregateSpec := spec.(*CountProcedureSpec)
+		if len(aggregateSpec.Columns) != 1 {
+			return nil, "", false
+		}
+		return aggregateWindowCount{}, aggregateSpec.Columns[0], true
+	case SumKind:
+		aggregateSpec := spec.(*SumProcedureSpec)
+		if len(aggregateSpec.Columns) != 1 {
+			return nil, "", false
+		}
+		return aggregateWindowSum{}, aggregateSpec.Columns[0], true
+	case MeanKind:
+		aggregateSpec := spec.(*MeanProcedureSpec)
+		if len(aggregateSpec.Columns) != 1 {
+			return nil, "", false
+		}
+		return aggregateWindowMean{}, aggregateSpec.Columns[0], true
+	default:
+		return nil, "", false
+	}
+}
+
+func (a AggregateWindowRule) isValidWindowSpec(spec *WindowProcedureSpec) bool {
+	return spec.TimeColumn == execute.DefaultTimeColLabel &&
+		spec.StartColumn == execute.DefaultStartColLabel &&
+		spec.StopColumn == execute.DefaultStopColLabel &&
+		spec.Window.Every != infinityVar.Duration()
+}
+
+type AggregateWindowCreateEmptyRule struct {
+	AggregateWindowRule
+}
+
+func (a AggregateWindowCreateEmptyRule) Name() string {
+	return "AggregateWindowCreateEmptyRule"
+}
+
+func (a AggregateWindowCreateEmptyRule) Pattern() plan.Pattern {
+	return plan.Pat(WindowKind,
+		plan.Pat(SchemaMutationKind,
+			plan.Pat(experimentaltable.FillKind,
+				plan.OneOf([]plan.ProcedureKind{MeanKind, SumKind, CountKind},
+					plan.Pat(WindowKind, plan.Any())))))
```

Comment:
What is the intent of this rule? I feel like there some context I'm missing. I thought that there was an issue with selectors where we needed to use `fill` to fill empty issues, but maybe I'm misremembering.